client/signup/steps/site-topic/index.jsx

```diff
@@ -91,9 +106,11 @@ class SiteTopicStep extends Component {
 	}
 }
 
-// TODO:
-// Connect to the real action creators and selectors.
 export default connect(
-	null,
-	{ setSiteTopic: () => {} }
+	state => ( {
+		siteTopic: getSignupStepsSiteTopic( state ),
+	} ),
+	{
+		setSiteTopic,
+	}
```

Comment:
In the segmentation step, we wrote this part in a way that the submit handlers are in a separate action, so that it would make things easier if we want to unit test a component(props to @ramonjd for that suggestion on that PR):
```
submitSiteTopic = event => {
		event.preventDefault();
        this.props.submitStep( this.state.siteTopicInputValue || '' );
	};
```

and the connectors will be:

```
export default connect(
	state => ( {
		siteTopic: getSignupStepsSiteTopic( state ),
	} ),
	( dispatch, ownProps ) => ( {
		submitStep: siteTopicInputValue => {
			dispatch( setSiteTopic( siteTopicInputValue ) );
			// Create site
			SignupActions.submitSignupStep(
				{
					processingMessage: i18n.translate( 'Collecting your information' ),
					stepName: ownProps.stepName,
				},
				[],
				{
					siteTopic: siteTopicInputValue,
				}
			);
			ownProps.goToNextStep( ownProps.flowName );
		},
	} )
)( localize( SiteTopicStep ) );
```

Reply:
Nice tip! I've adapted it :)